CMakeLists.txt

```diff
@@ -48,7 +48,13 @@ if (WIN32)
   # This allows us to pick up the correct version of Python instead of always picking the latest
   # installed on the machine.
   set(Python3_FIND_REGISTRY NEVER)
+  endif()
+
+# Use the Python version specified by pythonLocation.
+if(NOT DEFINED Python_ROOT_DIR AND DEFINED ENV{pythonLocation})
+  set(Python3_ROOT_DIR "$ENV{pythonLocation}")
```

Comment:
Why is this needed? find_package below should find it and set the root dir, right?

Reply:
So this change is needed just for the buildbot? Wouldn't it be better if we passed in the suggestion as a cmake variable?

e.g. `-DPython3_ROOT_DIR=...`

Reply:
That should work as well. The current code now allows for either passing `-DPython3_ROOT_DIR=...` via cmake or, if that is not defined, leveraging the `pythonLocation ` environment variable.  This is consistent with what is done by the third_party tools which has similar code: 

```
./onnx/third_party/pybind11/tools/pybind11NewTools.cmake:  if(NOT DEFINED Python_ROOT_DIR AND DEFINED ENV{pythonLocation})
./pybind11/tools/pybind11NewTools.cmake:  if(NOT DEFINED Python_ROOT_DIR AND DEFINED ENV{pythonLocation})
```

Reply:
Reading environment variables like this is generally frowned upon in cmake. Unless there is a strong reason to do it, it is better to rely on the cmake options.

Reply:
ok done